[IMP] inventory: overhaul 3rd party shippers (#8634)
odoo/documentation: content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst

Next, add the carrier details by going to :menuselection:`Inventory app --> Configuration -->
Shipping Methods`.

.. note::
   The list often includes **two** delivery methods from the same :guilabel:`Provider`: one for
   international and one for domestic shipping.

   Additional delivery methods can be created for specific purposes, such as :doc:`packaging
   <../../product_management/product_tracking/packaging>`.

.. seealso::
   :doc:`Configure delivery methods <../setup_configuration/delivery_method>`

The delivery method must be published before it can be used. To do so, choose the desired delivery
method.

.. image:: third_party_shipper/publish-delivery-method.png
   :align: center
   :alt: Display list of delivery methods and the "is Published" column.

On the shipping method page, configure the :doc:`delivery method details
<../setup_configuration/delivery_method>`, such as :guilabel:`Integration Level`,
:guilabel:`Delivery Product`, and :guilabel:`Invoicing Policy`.

Then, in the :guilabel:`Configuration` tab, input the API credentials and password with the shipping
carrier. Then, go to the :guilabel:`Destination Availability` tab and configure the service coverage

> of the shipping carrier.

Reviewer: For these instructions here, do you think some additional instruction on what these are, how you find them, and what they affect would be helpful? I know it's different per carrier, but maybe a .. note:: or a .. tip:: saying "read the doc for individual carriers to find this stuff" would be nice

> of the shipping carrier.

Author: ```suggestion
of the shipping carrier.  For specific details about each
shipping method, refer to the following documents:

.. seealso::
   - :doc:`DHL credentials <dhl_credentials>`
   - :doc:`Sendcloud credentials <sendcloud_shipping>`
   - :doc:`UPS credentials <ups_credentials>
```

Publish
~~~~~~~

With the delivery method details configured, click the :guilabel:`Test Environment` smart button to
activate it.

.. warning::

>    Set the delivery method to :guilabel:`Production` charges customers for shipping. Verify all

Reviewer: ```suggestion
   Setting the delivery method to :guilabel:`Production` charges customers for shipping. Verify all
```

>    configurations before launching the delivery method to :guilabel:`Production`.

Reviewer: ```suggestion
   configurations are correct before launching the delivery method to :guilabel:`Production`.
```

      :alt: Display "Unpublished" smart button at the top of the delivery method page.

.. _inventory/shipping_receiving/configure-source-address:

Source address
--------------

Configure the source address for package shipments by navigating to :menuselection:`Inventory app
--> Configuration --> Warehouses` and selecting the desired warehouse.

On the warehouse configuration page, hover over the :guilabel:`Address` field and select the
:guilabel:`➡️ (right arrow)` icon. Doing so opens the warehouse contact page.

.. image:: third_party_shipper/internal-link.png
   :align: center
   :alt: Show the internal link icon that appears when hovering over "Address".

Ensure the warehouse's :guilabel:`Address` and :guilabel:`Phone` number are entered accurately, as
they are used to calculate cost of shipping.

.. image:: third_party_shipper/company.png
   :align: center
   :alt: Show company address and phone number.

.. _inventory/shipping_receiving/configure-weight:

Product weight
--------------

To calculate the price of shipping, specify the weight of products by going to
:menuselection:`Inventory app --> Products --> Products`, and selecting the desired product.

> Then, switch to the :guilabel:`Inventory` tab, define the :guilabel:`Weight` of the product, in

Reviewer: ```suggestion
Then, switch to the :guilabel:`Inventory` tab, and define the :guilabel:`Weight` of the product, in
```

> kilograms.

Reviewer: Why kilograms? Does it have to be? Can different products have different UoMs?

Author: You're right! I confirmed with an expert, and this field has become really flexible since 8 years ago, when the line was written (:

.. image:: third_party_shipper/product-weight.png
   :align: center
   :alt: Display the "Weight" field in the Inventory tab of the product form.

.. tip::
   Make sure to convert weights into kilograms for the calculations to work properly.

Apply third-party shipping carrier on sales order
=================================================

After configuring the third-party carrier's :ref:`delivery method
<inventory/shipping_receiving/configure-delivery-method>` in Odoo, navigate to a quotation by going

> to :menuselection:`Sales app --> Orders --> Quotations`.

Reviewer: Or, we can create a new quotation also right?

Add shipping
------------

Add the cost of shipping through a third-party carrier to a quotation by clicking the :guilabel:`Add
Shipping` button in the bottom-right corner.

.. image:: ../advanced_operations_shipping/invoicing/add-shipping.png
   :align: center
   :alt: Show the "Add shipping" button at the bottom of a quotation.

> In the pop-up window, select the intended carrier from the :guilabel:`Shipping Method` drop-down

Reviewer: ```suggestion
In the resulting pop-up window, select the intended carrier from the :guilabel:`Shipping Method` drop-down
```

menu. The :guilabel:`Cost` field is automatically filled based on the :ref:`product weight
<inventory/shipping_receiving/configure-weight>` and distance between the warehouse's :ref:`source
address <inventory/shipping_receiving/configure-source-address>` and the customer address.

Click :guilabel:`Get Rate` to add the cost to the quotation, which is listed as the delivery product
configured earlier.

.. image:: ../advanced_operations_shipping/invoicing/add-a-shipping-method.png
   :align: center
   :alt: Show the "Get rate" button in the "Add a shipping method" pop-up.

.. seealso::
   :doc:`Charge customers for shipping after product delivery
   <../advanced_operations_shipping/invoicing>`

Delivery information
--------------------

After confirming the sales order, click the :guilabel:`Delivery` smart button to view the linked
delivery order.

In the :guilabel:`Additional info` tab, the third-party shipping carrier is listed in the

> :guilabel:`Carrier` field. When the delivery method set to :ref:`production mode

Reviewer: ```suggestion
:guilabel:`Carrier` field. When the delivery method is set to :ref:`production mode
```

==============================
Integrate third-party carriers
==============================

.. _inventory/shipping/third_party:

Link third-party shipping carriers to Odoo databases to verify carriers' delivery to specific
addresses, :doc:`automatically calculate shipping costs <delivery_method>`, and :doc:`generate

> shipping labels <../advanced_operations_shipping/labels>`.

Author: Very nice rewrite, thank you!

Delivery method
---------------

To configure the API credentials and activate the shipping carrier, begin by going to
:menuselection:`Inventory app --> Configuration --> Shipping Methods`, and select the desired
delivery method.

.. note::
   The list often includes **two** delivery methods from the same :guilabel:`Provider`: one for
   international shipping and one for domestic shipping.

   Additional delivery methods can be created for specific purposes, such as :doc:`packaging
   <../../product_management/product_tracking/packaging>`.

.. seealso::
   :doc:`Configure delivery methods <delivery_method>`

.. note::
   Ensure the delivery method :guilabel:`Is Published` when it should be available on the *Website*
   app. To publish a delivery method on the website, click the desired delivery method and click the
   :guilabel:`Unpublished` smart button.

The :guilabel:`Shipping Method` page contains details about the provider, including:

- :guilabel:`Shipping Method` (*Required field*): the name of the delivery method (e.g. `FedEx US`,
  `FedEx EU`, etc.).
- :guilabel:`Provider` (*Required field*): choose the delivery service, like Fedex.

> - :guilabel:`Integration Level`: choose :guilabel:`Get Rate` to simply get an :ref:`estimated

Author: These are good points! Taking note for myself here: upon selecting a Provider, the following fields appear:

Integration Level
Invoicing Policy
Insurance Percentage
Configuration tab

  drop-down menu. Leave the field blank to apply the method to all companies.
- :guilabel:`Delivery Product` (*Required field*): the delivery charge name that's added to the
  :abbr:`SO (Sales Order)` or invoice.
- :guilabel:`Free if order amount is above`: checking this box enables free shipping if the customer
  spends above the specified amount.

.. figure:: third_party_shipper/fedex-shipping-config.png
   :align: center
   :alt: Screenshot of a FedEx shipping method.

   **Shipping Method** configuration page for `FedEx US`.

In the :guilabel:`Configuration` tab, fill out the API credential fields (e.g. API key, password,
account number). Depending on the third party shipping carrier, the :guilabel:`Configuration` will
contain different required fields. For more details about configuring specific carriers'
credentials, refer to the following documents:

.. seealso::
   - :doc:`DHL credentials <dhl_credentials>`
   - :doc:`Sendcloud credentials <sendcloud_shipping>`
   - :doc:`UPS credentials <ups_credentials>`

Publish
~~~~~~~

With the delivery method details configured, click the :guilabel:`Test Environment` smart button to

> set it to production environment.

Author: ```suggestion
set it to :guilabel:`Production Environment`.
```

   :align: center
   :alt: Display the "Weight" field in the Inventory tab of the product form.

Apply third-party shipping carrier
==================================

Shipping carriers can be applied on a :abbr:`SO (Sales Order)`, invoice, or delivery order.

After configuring the third-party carrier's :ref:`delivery method
<inventory/shipping_receiving/configure-delivery-method>` in Odoo, create or navigate to a quotation
by going to :menuselection:`Sales app --> Orders --> Quotations`.

.. _inventory/shipping_receiving/third-party-so:

Sales order
-----------

To assign a third party shipping carrier and get an estimated cost of shipping, begin by going to
:menuselection:`Sales app --> Orders --> Quotations`. Create a select an existing quotation, and add
the cost of shipping through a third-party carrier to a quotation by clicking the :guilabel:`Add
Shipping` button in the bottom-right corner.

.. image:: ../advanced_operations_shipping/invoicing/add-shipping.png
   :align: center
   :alt: Show the "Add shipping" button at the bottom of a quotation.

In the resulting pop-up window, select the intended carrier from the :guilabel:`Shipping Method`

> drop-down menu. The :guilabel:`Cost` field is automatically filled based on the :ref:`product weight
> <inventory/shipping_receiving/configure-weight>` and distance between the warehouse's :ref:`source

Author: ```suggestion
drop-down menu. The :guilabel:`Cost` field is automatically filled based on:
- the amount specified in the :guilabel:`Total Order Weight` field (if it is not provided, the sum of :ref:`product weights <inventory/shipping_receiving/configure-weight>` in the order is used) 
- the distance between the warehouse's :ref:`source
```

address <inventory/shipping_receiving/configure-source-address>` and the customer address.

.. _inventory/shipping_receiving/third-party-rate:

> Click :guilabel:`Get Rate` in the :guilabel:`Add a shipping method` pop-up window to get the

Author: ```suggestion
After selecting a third party provider in the :guilabel:`Shipping Method` field, click :guilabel:`Get Rate` in the :guilabel:`Add a shipping method` pop-up window to get the
```

estimated cost through the shipping connector. Then, click the :guilabel:`Add` button to add the
delivery charge to the :abbr:`SO (Sales Order)` or invoice.

.. image:: ../advanced_operations_shipping/invoicing/add-a-shipping-method.png
   :align: center
   :alt: Show the "Get rate" button in the "Add a shipping method" pop-up.

.. seealso::
   :doc:`Charge customers for shipping after product delivery
   <../advanced_operations_shipping/invoicing>`

.. _inventory/shipping_receiving/third-party-do:

Delivery order
--------------

For users making shipments without installing the *Sales* app, assign the shipping carrier to the

> delivery order, by first going to the :menuselection:`Inventory` app. Select the :guilabel:`Delivery

Reviewer 2: ```suggestion
delivery order, by first going to the :menuselection:`Inventory` app. Then, from the :guilabel:`Inventory Overview` dashboard, select the :guilabel:`Delivery
```

> Orders` operation type, and choose the desired delivery order.

Reviewer 2: do you think it would be helpful to mention that the user should choose a DO that isn't already marked as 'Done' -- or do you think that is redundant info?

> Orders` operation type, and choose the desired delivery order.

Author: ```suggestion
Orders` operation type, and choose the desired delivery order that is not already marked as :guilabel:`Done` or :guilabel:`Cancelled`.
```

> In the :guilabel:`Additional info` tab, set :guilabel:`Carrier` field to the third-party shipping

Reviewer 2: ```suggestion
In the :guilabel:`Additional info` tab, set :guilabel:`Carrier` field to the desired third-party shipping
```

==============================
Integrate third-party carriers
==============================

.. |SO| replace:: :abbr:`SO (Sales Order)`
.. |DO| replace:: :abbr:`DO (Delivery Order)`

.. _inventory/shipping/third_party:

Users can link third-party shipping carriers to Odoo databases, in order to verify carriers'
delivery to specific addresses, :doc:`automatically calculate shipping costs <delivery_method>`, and
:doc:`generate shipping labels <../advanced_operations_shipping/labels>`.

In Odoo, shipping carriers can be applied to a sales order (SO), invoice, or delivery order (DO).
The following is a list of available shipping connectors:

.. list-table::
   :header-rows: 1
   :stub-columns: 1

   * - Carrier
     - Region availability
   * - FedEx
     - All
   * - :doc:`DHL <dhl_credentials>`
     - All
   * - :doc:`UPS <ups_credentials>`
     - All
   * - US Postal Service
     - United States of America
   * - :doc:`Sendcloud <sendcloud_shipping>`
     - EU
   * - Bpost
     - Belgium
   * - Easypost
     - North America
   * - Shiprocket
     - India

Configuration
=============

> To ensure proper setup of the third-party shipping carrier with Odoo, follow these steps:

Reviewer 2: ```suggestion
To ensure proper setup of a third-party shipping carrier with Odoo, follow these steps:
```

  the senders if the package is lost or stolen in transit.

.. figure:: third_party_shipper/fedex.png
   :align: center
   :alt: Screenshot of a FedEx shipping method.

   **Shipping Method** configuration page for `FedEx US`.

In the :guilabel:`Configuration` tab, fill out the API credential fields (e.g. API key, password,
account number, etc.). Depending on the third-party shipping carrier chosen in the
:guilabel:`Provider` field, the :guilabel:`Configuration` tab will contain different required
fields. For more details about configuring specific carriers' credentials, refer to the following
documents:

.. seealso::
   - :doc:`DHL credentials <dhl_credentials>`
   - :doc:`Sendcloud credentials <sendcloud_shipping>`
   - :doc:`UPS credentials <ups_credentials>`

.. _inventory/shipping_receiving/production-env:

Production environment
----------------------

With the delivery method details configured, click the :guilabel:`Test Environment` smart button to
set it to :guilabel:`Production Environment`.

.. warning::
   Setting the delivery method to :guilabel:`Production` creates **real** shipping labels, and users

>    are at risk for being charged through their carrier account (e.g. UPS, FedEx, etc.), **before**

Reviewer 2: ```suggestion
   are at risk of being charged through their carrier account (e.g. UPS, FedEx, etc.) **before**
```

   users charge customers for shipping. Verify all configurations are correct before launching the
   delivery method to :guilabel:`Production`.

.. image:: third_party_shipper/production.png
   :align: center
   :alt: Show the "Test Environment" smart button.

.. _inventory/shipping_receiving/configure-source-address:

Warehouse configuration
-----------------------

Ensure the warehouse's :guilabel:`Address` (including ZIP code) and :guilabel:`Phone` number are
entered accurately. To do that, go to :menuselection:`Inventory app --> Configuration -->
Warehouses`, and select the desired warehouse.

On the warehouse configuration page, open the warehouse contact page by clicking the
:guilabel:`Company` field.

.. image:: third_party_shipper/internal-link.png
   :align: center
   :alt: Highlight the "Company" field.

> Verify the :guilabel:`Address` and :guilabel:`Phone` number are correct, as they are required for

Reviewer 2: ```suggestion
Verify that the :guilabel:`Address` and :guilabel:`Phone` number are correct, as they are required for
```

the shipping connector to work properly.

.. image:: third_party_shipper/company.png
   :align: center
   :alt: Show company address and phone number.

.. _inventory/shipping_receiving/configure-weight:

Product weight
--------------

For the carrier integration to work properly, specify the weight of products by going to
:menuselection:`Inventory app --> Products --> Products`, and selecting the desired product.

Then, switch to the :guilabel:`Inventory` tab, and define the :guilabel:`Weight` of the product in
the :guilabel:`Logistics` section.

.. image:: third_party_shipper/product-weight.png
   :align: center
   :alt: Display the "Weight" field in the Inventory tab of the product form.

Apply third-party shipping carrier
==================================

Shipping carriers can be applied on a :abbr:`SO (Sales Order)`, invoice, or delivery order.

After configuring the third-party carrier's :ref:`delivery method
<inventory/shipping_receiving/configure-delivery-method>` in Odoo, create or navigate to a quotation
by going to :menuselection:`Sales app --> Orders --> Quotations`.

.. _inventory/shipping_receiving/third-party-so:

Sales order
-----------

> To assign a third party shipping carrier, and get an estimated cost of shipping, begin by going to

Reviewer 2: ```suggestion
To assign a third-party shipping carrier, and get an estimated cost of shipping, begin by going to
```

:menuselection:`Sales app --> Orders --> Quotations`. Create or select an existing quotation, and
add the cost of shipping through a third-party carrier to a quotation, by clicking the
:guilabel:`Add Shipping` button in the bottom-right corner of the :guilabel:`Order Lines` tab.

.. image:: ../advanced_operations_shipping/invoicing/add-shipping.png
   :align: center
   :alt: Show the "Add shipping" button at the bottom of a quotation.

In the resulting :guilabel:`Add a shipping method` pop-up window, select the intended carrier from
the :guilabel:`Shipping Method` drop-down menu. The :guilabel:`Cost` field is automatically filled
based on:

- the amount specified in the :guilabel:`Total Order Weight` field (if it is not provided, the sum
  of :ref:`product weights <inventory/shipping_receiving/configure-weight>` in the order is used)
- the distance between the warehouse's :ref:`source address
  <inventory/shipping_receiving/configure-source-address>` and the customer's address.

.. _inventory/shipping_receiving/third-party-rate:

> After selecting a third party provider in the :guilabel:`Shipping Method` field, click

Reviewer 2: ```suggestion
After selecting a third-party provider in the :guilabel:`Shipping Method` field, click
```

:guilabel:`Get Rate` in the :guilabel:`Add a shipping method` pop-up window to get the estimated
cost through the shipping connector. Then, click the :guilabel:`Add` button to add the delivery
charge to the :abbr:`SO (Sales Order)` or invoice.

.. image:: ../advanced_operations_shipping/invoicing/add-a-shipping-method.png
   :align: center
   :alt: Show the "Get rate" button in the "Add a shipping method" pop-up.

.. seealso::
   :doc:`Charge customers for shipping after product delivery
   <../advanced_operations_shipping/invoicing>`

.. _inventory/shipping_receiving/third-party-do:

Delivery order
--------------

For users making shipments without installing the *Sales* app, assign the shipping carrier to the
delivery order, by first going to the :menuselection:`Inventory` app. Then, from the
:guilabel:`Inventory Overview` dashboard, select the :guilabel:`Delivery Orders` operation type, and
choose the desired delivery order that is not already marked as :guilabel:`Done` or
:guilabel:`Cancelled`.

> In the :guilabel:`Additional info` tab, set :guilabel:`Carrier` field to the desired third-party

Reviewer 2: ```suggestion
In the :guilabel:`Additional info` tab, set the :guilabel:`Carrier` field to the desired third-party
```
